--- /tmp/cabal2spec	2012-01-21 12:54:23.608310800 +0900
+++ /usr/bin/cabal2spec	2012-01-21 12:51:26.850308148 +0900
@@ -61,12 +61,14 @@
   echo "Warning: tarball version ($TARVERSION) and cabal version ($VERSION) differ!"
 fi
 
+LICENSE=$(sed -n "s/^[Ll]icense[ \t]*:[ \t]*//p" $CABAL)
+DATE=$(env LANG=C date +"%a %b %e %Y")
+
 CABALFILENAME=$(basename $CABAL .cabal)
 if [ "$CABALFILENAME" != "$NAME" ]; then
     echo "Warning: .cabal filename ($CABALFILENAME) and cabal Name field ($NAME) differ!"
 fi
 
-
 if grep -v ^-- $CABAL | grep -qi exposed-modules:; then
     HAS_LIB=yes
 fi
@@ -99,8 +101,15 @@
   TEMPLATEDIR=/etc/rpmdevtools
 fi
 
-cp ${TEMPLATEDIR}/${TEMPLATE} $SPECFILE
+#cp ${TEMPLATEDIR}/${TEMPLATE} $SPECFILE
+
+#sed -i -e "s/@PACKAGE@/$NAME/g" -e "s/@VERSION@/$VERSION/g" -e "s/@DATE@/$(env LANG=C date +"%a %b %e %Y")/" $SPECFILE
 
-sed -i -e "s/@PACKAGE@/$NAME/g" -e "s/@VERSION@/$VERSION/g" -e "s/@DATE@/$(env LANG=C date +"%a %b %e %Y")/" $SPECFILE
+sed -e "
+s/@PACKAGE@/$NAME/g
+s/@VERSION@/$VERSION/g
+s/@LICENSE@/$LICENSE/g
+s/@DATE@/$DATE/g
+" ${TEMPLATEDIR}/${TEMPLATE} > $SPECFILE
 
 #echo "created $SPECFILE (${$FORCE_LIB:-${HAS_BIN:+bin}}${HAS_LIB:+lib}) for $NAME-$VERSION"
